testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/broker/OidcUserInfoClaimToRoleMapperTest.java

```diff
@@ -0,0 +1,95 @@
+package org.keycloak.testsuite.broker;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.junit.Test;
+import org.keycloak.admin.client.resource.IdentityProviderResource;
+import org.keycloak.broker.oidc.mappers.ClaimToRoleMapper;
+import org.keycloak.broker.provider.ConfigConstants;
+import org.keycloak.models.IdentityProviderMapperModel;
+import org.keycloak.models.IdentityProviderMapperSyncMode;
+import org.keycloak.representations.idm.IdentityProviderMapperRepresentation;
+import org.keycloak.representations.idm.IdentityProviderRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+
+import java.util.List;
+
+public class OidcUserInfoClaimToRoleMapperTest extends AbstractRoleMapperTest {
+
+    private static final String USER_INFO_CLAIM = KcOidcBrokerClientUserInfoTest.ATTRIBUTE_TO_MAP_USER_INFO;
+    private static final String USER_INFO_CLAIM_VALUE = "value 1";
+    private String claimOnSecondLogin = "";
+
+    @Override
+    protected BrokerConfiguration getBrokerConfiguration() {
+        return new KcOidcBrokerClientUserInfoTest().getBrokerConfiguration();
```

Comment:
I think you can remove the class "KcOidcBrokerClientUserInfoTest" entirely and move the class KcOidcBrokerClientUserInfoTest.KcOidcBrokerConfigurationUserInfoOnlyMappers as the private method to this class? Also move the constant ATTRIBUTE_TO_MAP_USER_INFO from that class directly to your "OidcUserInfoClaimToRoleMapperTest"